### ISSUE TITLE
Support for historical countries?

### ISSUE THREAD

**Issue by jesperronn** [countries/countries#671]
Thanks for this gem. It looks really useful. 

Do you have any plans to add historical countries as well to this list, for instance Czechoslovakia or ex Yugoslavia, or ex Soviet union?

If not, what are the plans going forward if a country splits or dissolves in the future?

**Comment by rposborne** [comment]
So far we have simply not modeled historical data, I don't think it's an unreasonable request and most likely be solved with tracking a "dissolved_on" value. 

Historically we delete these records.

**Comment by dima4p** [comment]
Yes, I also need the project to have the countries like DDR, Yugoslavia, USSR, etc.

**Comment by hiSandog** [comment]
Historical countries would probably be safest as a separate dataset or opt-in API rather than being mixed into the current ISO-style country list. The useful fields are different: validity date ranges, successor/predecessor relationships, aliases, and whether codes are historical ISO values or project-defined identifiers. Keeping that boundary explicit would avoid surprising users who expect the existing list to represent current countries only.

**Comment by dima4p** [comment]
> Historical countries would probably be safest as a separate dataset or opt-in API rather than being mixed into the current ISO-style country list. The useful fields are different: validity date ranges, successor/predecessor relationships, aliases, and whether codes are historical ISO values or project-defined identifiers. Keeping that boundary explicit would avoid surprising users who expect the existing list to represent current countries only.

Nothing can prevent us to add a parameter to the #search to show old countries too.
